### DISCUSSION TITLE
What is the size limit of the vtk files that can be uploaded?

### DISCUSSION THREAD

**Discussion by DanielSevilla2897** [Kitware/trame#49] [Trame/VTK] [answered]
Hello,
I am trying to display a vtk file in TRAME, but as it has 1577910 points (385 MB) I can’t open it. Do you know of any way to display them?

Thank you.

**Comment by jourdain** [comment, marked as answer]
Hi @DanielSevilla2897,

Could you explain what you've done as 400 MB is rather small for trame/vtk rendering. 

Are you uploading it using the file infrastructure of trame or simply reading it from disk directly in Python? 
If you are uploading it, then yes the basic implementation will send your file over one websocket message and it will likely fail due to the 4KB limit defined by default. You can bump that value if you are in a trusted environment by defining the following environment variable.

```bash
export WSLINK_MAX_MSG_SIZE=1000000000    # 1GB
``` 

Again this is just speculation, as I'm not sure what you've been doing and if you are trying remote or local rendering...

HTH,

Seb

**Reply by DanielSevilla2897** [reply]
Hi Jourdain,

thank you for the answer.
We are trying to open a vtk (unstructured grid) and we are having some problems when rendering. It turns out that it doesn't open the internal mesh, only the external one. Do you know how we could fix it?

**Reply by jourdain** [reply]
VTK always render the external mesh so I'm not sure what is the issue here? Maybe if you could truly explain what you are doing, what you are seeing and what you would like to see, we might be able to help you. With none of that, it is impossible to provide any meaningful guidance.
